src/shared/bot_nav_shared.h

```diff
@@ -79,16 +79,32 @@ enum navPolyAreas
 
 // Part of header which must contain 4-byte members only
 struct NavgenConfig {
-	float requestedCellHeight;
-	float stepSize;
-	int excludeCaulk; // boolean - exclude caulk surfaces
-	int excludeSky; // boolean - exclude surfaces with surfaceparm sky from navmesh generation
-	int filterGaps; // boolean - add new walkable spans so bots can walk over small gaps
-	int generatePatchTris; // boolean - generate triangles from the BSP's patches
-	float autojumpSecurity; // percentage - allow to use part of jump magnitude (with default gravity of 800) as stepsize. The result can not excess the agent's height, except if STEPSIZE is already doing it (then STEPSIZE will be used)
-	bool crouchSupport = true;
-
-	static NavgenConfig Default() { return { 2.0f, STEPSIZE, 1, 1, 1, 1, 0.5f }; }
+	float requestedCellHeight = 2.f;
+	float stepSize = STEPSIZE;
+	float autojumpSecurity = 0.5f; // percentage - allow to use part of jump magnitude (with default gravity of 800) as stepsize. The result can not excess the agent's height, except if STEPSIZE is already doing it (then STEPSIZE will be used)
+	bool excludeCaulk = 1; // exclude caulk surfaces
+	bool excludeSky = 1; // exclude surfaces with surfaceparm sky from navmesh generation
+	bool filterGaps = 1; // add new walkable spans so bots can walk over small gaps
+	bool generatePatchTris = 1; // generate triangles from the BSP's patches
+	bool crouchSupport = true; // use crouchMaxs.z value instead of Maxs.z to compute paths
+
+	bool operator==( NavgenConfig const& other ) const
+	{
+		return requestedCellHeight == other.requestedCellHeight
+			&& stepSize == other.stepSize
+			&& excludeCaulk == other.excludeCaulk
```

Comment:
Why should we have to write this boilerplate?

Reply:
just FYI I disagree with both of you here. This code is safer because less fragile from padding.
This whole struct is very fragile anyway since there is no way to check the size of the header. To me, it's poorly engineered, and I'd rather have boilerplate than loosing hours wondering why code does not work because I introduced a boolean or a short, which triggers paddings and fucks the memcmp.

Reply:
@slipher is the code correct even if that's not the way you would do it?

Reply:
> @slipher is the code correct even if that's not the way you would do it?

No because it breaks the endianness swapping stuff. As I said elsewhere in the thread, we could consider getting rid of that stuff, but that should be done intentionally, rather than by letting it fall into disrepair.